functions/form-payment-reconciliation/index.js

```diff
@@ -0,0 +1,291 @@
+ * - SLACK_API_SECRET: the API key for posting reports to slack
+ */
+
+const { SSMClient, GetParameterCommand } = require('@aws-sdk/client-ssm')
+const { generateLinkToCloudwatch, getPostToSlack } = require('./src/utils')
+
+const AWS_REGION = process.env.AWS_REGION
+
+const ENV_SITE_NAME = process.env.SSM_ENV_SITE_NAME
+const CRON_API_PREFIX = `https://${
+  ENV_SITE_NAME === 'prod' ? '' : `${ENV_SITE_NAME}.`
+}form.gov.sg/api/v3/payments/reconcile/`
+
+const PARAMETER_STORE_NAME = `${ENV_SITE_NAME}-cron-payment`
+const CRON_PAYMENT_API_SECRET_KEY = 'CRON_PAYMENT_API_SECRET'
+const CRON_PAYMENT_SLACK_SECRET_KEY = 'SLACK_API_SECRET'
+
+const API_AUTH_HEADER = 'x-formsg-cron-payment-secret'
+
+const MAX_AGE_HRS_EVENTS = 7
+
+/**
+ * Helper function to obtain secrets map from parameter store.
+ */
+const getSSMSecrets = () => {
+  // Regex used to match key-value pairs from parameter store.
+  // Makes use of capture groups to obtain the key and value.
+  const KEY_VALUE_PAIR_REGEX = /^([^\s=]+)\s*=\s*(\S+)$/
+
+  const awsSsmClient = new SSMClient({ region: AWS_REGION })
+  const input = {
+    Name: PARAMETER_STORE_NAME,
+    WithDecryption: true,
+  }
+  const command = new GetParameterCommand(input)
+  return awsSsmClient.send(command).then((res) => {
+    const parametersAsString = res.Parameter.Value
+    const parameters = parametersAsString
+      .split(/[\r\n]+/)
+      .map((line) => line.trim())
+      .filter((line) => !!line) // Filter out empty lines
+    return parameters.reduce((parameterMap, line) => {
+      const match = line.match(KEY_VALUE_PAIR_REGEX)
+      if (match && match.length === 3) parameterMap[match[1]] = match[2]
+      return parameterMap
+    }, {})
+  })
+}
+
+/**
+ * Helper function to get API functions given the API secret.
+ */
+const getApi = (apiSecret) => {
+  const getIncompletePayments = async () => {
+    return fetch(`${CRON_API_PREFIX}incompletePayments`, {
+      headers: { [API_AUTH_HEADER]: apiSecret },
+    }).then(async (res) => ({ ok: res.ok, data: await res.json() }))
+  }
+
+  const reconcileAccount = async (stripeAccount, paymentIds) => {
+    return fetch(
+      `${CRON_API_PREFIX}account/${stripeAccount}?maxAgeHrs=${MAX_AGE_HRS_EVENTS}`,
+      {
+        method: 'POST',
+        headers: {
+          [API_AUTH_HEADER]: apiSecret,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ paymentIds }),
+      },
+    ).then(async (res) => ({ ok: res.ok, data: await res.json() }))
+  }
+
+  return { getIncompletePayments, reconcileAccount }
+}
+
+async function main(events, context) {
+  // Things will be appended as we go along.
+  const REPORT = [
+    `*Reconciliation report for ${ENV_SITE_NAME}*`,
+    '---',
+    '*Critical messages:*',
+  ]
+
+  console.log('CRON job starting')
+  const CRON_START_TIME = Date.now()
+
+  // Step 0(a): Load secrets from SSM
+  const secret = await getSSMSecrets()
+
```

Comment:
we will know about the shared secret since we will receive a whole bunch of failed requests in our reconciliation report. For missing slack secret key, there's no way to really notify us unless we set up an entirely different flow e.g. by email or something (otherwise if the lambda has no access to slack, it can't notify us via slack!). the only way is to notice that the lambda didnt run at xx:(0|3)0.